dfsclient/src/main/java/org/hpccsystems/dfs/client/HPCCRecordBuilder.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * ##############################################################################
+ * 
+ * HPCC SYSTEMS software Copyright (C) 2018 HPCC Systems®.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ * 
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ * ##############################################################################
+ */
+
+package org.hpccsystems.dfs.client;
+
+import org.hpccsystems.dfs.client.IRecordBuilder;
+import org.hpccsystems.dfs.client.HPCCRecord;
+
+import org.hpccsystems.commons.ecl.FieldDef;
+import org.hpccsystems.commons.ecl.FieldType;
+
+public class HPCCRecordBuilder implements IRecordBuilder
+{
+    private Object[]            fields              = null;
+    private FieldDef            fieldDef            = null;
+
+    private HPCCRecordBuilder[] childRecordBuilders = null;
+
+    public HPCCRecordBuilder()
+    {
+    }
+
+    public void setRecordDefinition(FieldDef fieldDef)
```

Comment:
this might be related to a comment made in the commons pull request, but it seems a recorddefinition would require either a set of fielddefs, or FieldDef is actually a recdef

Reply:
FieldDef can represent any ECL field type. In this case the FieldDef has to be a record.